### PR TITLE
Rho now responds to OPTION requests. Choosing to simply handle it directly

### PR DESCRIPTION
**Pull request by kryptt**
This changes the behaviour from before, but I don't think anyone was handling options alongside this middleware..

### CODE REVIEW

**Comment by bryce-anderson** [comment]
By 'rho now response to OPTIONS requests', do you mean the `MethodNotAllowed` response served when querying an existing route with the wrong method? It doesn't explicitly reply to an OPTIONS with a normal status but maybe that is something rho should be doing...

At any rate, this looks fine to me. :+1:

**Comment by kryptt** [comment]
Yes, Rho now responds with MethodNotAllowed for OPTIONS.

I would rather the CORS middleware did not have to take over the OPTIONS request entirely since an arbitrary HttpService could choose to add a meaningful body to the OPTIONS method  (see: http://zacstewart.com/2012/04/14/http-options-method.html).

How would you suggest we handle it? Have CORS emit a warning when the status != 2** since it would otherwise respond with 200?

**Comment by bryce-anderson** [comment]
I have little experience with CORS so I'm not really the one to ask. I'm wondering (and this is probably a discussion for the rho issue tracker) if rho should _not_ return a `MethodNotAllowed` for an `OPTIONS` request?

File: server/src/main/scala/org/http4s/server/middleware/CORS.scala [hunk 1]
```diff
@@ -61,9 +60,14 @@ class CORS(service: HttpService, config: CORSConfig) extends CORS.CORSF {
...
 
   def options(req: Request, origin: Header, acrm: Header): Task[Option[Response]] =
-    service
-      .or(req, now(Response(Ok))).liftM[OptionT]
-      .map(corsHeaders(origin.value, acrm.value)).run
+    service.or(req, Task(Response(Ok))).liftM[OptionT].map { resp =>
+      if (resp.status.isSuccess)
+        corsHeaders(origin.value, acrm.value)(resp)
+      else {
+        logger.info(s"CORS headers would have been alloed for ${req.method} ${req.uri}")
```
Comment (bryce-anderson) [inline server/src/main/scala/org/http4s/server/middleware/CORS.scala]: Small typo in 'alloed'.

Comment (kryptt) [inline server/src/main/scala/org/http4s/server/middleware/CORS.scala]: thanks, fixed.

**Pull request closed without merging**